dbms/src/Flash/Mpp/MPPTunnelSet.cpp

```diff
@@ -26,83 +26,52 @@ namespace
 void checkPacketSize(size_t size)
 {
     static constexpr size_t max_packet_size = 1u << 31;
-    if (size >= max_packet_size)
+    if (unlikely(size >= max_packet_size))
         throw Exception(fmt::format("Packet is too large to send, size : {}", size));
 }
 
+TrackedMppDataPacketPtr serializePacket(tipb::SelectResponse & response)
+{
+    auto tracked_packet = std::make_shared<TrackedMppDataPacket>();
+    tracked_packet->serializeByResponse(response);
+    checkPacketSize(tracked_packet->getPacket().ByteSizeLong());
+    return tracked_packet;
+}
 } // namespace
 
 template <typename Tunnel>
-void MPPTunnelSetBase<Tunnel>::clearExecutionSummaries(tipb::SelectResponse & response)
+void MPPTunnelSetBase<Tunnel>::sendExecutionSummary(tipb::SelectResponse & response)
 {
-    /// can not use response.clear_execution_summaries() because
-    /// TiDB assume all the executor should return execution summary
-    for (int i = 0; i < response.execution_summaries_size(); i++)
-    {
-        auto * mutable_execution_summary = response.mutable_execution_summaries(i);
-        mutable_execution_summary->set_num_produced_rows(0);
-        mutable_execution_summary->set_num_iterations(0);
-        mutable_execution_summary->set_concurrency(0);
-    }
+    RUNTIME_CHECK(!tunnels.empty());
+    // for execution summary, only need to send to one tunnel.
+    tunnels[0]->write(serializePacket(response));
 }
 
 template <typename Tunnel>
 void MPPTunnelSetBase<Tunnel>::write(tipb::SelectResponse & response)
```

Comment:
Since there is `broadcastWrite` and `partitionWrite`, I think it's better to rename `write` to `passThroughWrite`

Reply:
Now both passThrough exchange also run on `BroadcastOrPassThroughWriter` or not `StreamingWriter`, because considering that the future local tunnel will pass the block directly.
`broadcastWrite` has renamed to `broadcastOrPassThroughWrite`.